c/cert/src/rules/ENV30-C/DoNotModifyTheReturnValueOfCertainFunctions.ql

```diff
@@ -0,0 +1,68 @@
+/**
+ * @id c/cert/do-not-modify-the-return-value-of-certain-functions
+ * @name ENV30-C: Do not modify the return value of certain functions
+ * @description Return value of getenv and similar functions cannot be modified.
+ * @kind path-problem
+ * @precision very-high
+ * @problem.severity error
+ * @tags external/cert/id/env30-c
+ *       correctness
+ *       external/cert/obligation/rule
+ */
+
+import cpp
+import codingstandards.c.cert
+import semmle.code.cpp.dataflow.DataFlow
+import DataFlow::PathGraph
+
+/*
+ * Call to functions that return pointers to environment objects that should not be modified.
+ */
+
+class NotModifiableCall extends FunctionCall {
+  NotModifiableCall() {
+    this.getTarget()
+        .hasGlobalOrStdName(["getenv", "setlocale", "localeconv", "asctime", "strerror"])
+  }
+}
+
+/*
+ * An expression that modifies an object.
+ */
+
+class ObjectWrite extends Expr {
```

Comment:
hm, can we double check if there is anything already written to describe this?
for example maybe something in the SideEffects libs? Although I dont quite get why the[ C specific lib](https://github.com/github/codeql-coding-standards/blob/main/c/common/src/codingstandards/c/SideEffects.qll) for that is all private. or maybe something close to [this](https://github.com/github/codeql-coding-standards/blob/main/cpp/common/src/codingstandards/cpp/SideEffect.qll#L233)?

Reply:
makes sense! at this point then totally fair that it doesnt make sense to expose as smaller simpler c specific write lib, if it comes up again then we can but otherwise good as is :)